python/src/trezorlib/ethereum.py

```diff
@@ -179,35 +199,43 @@ def sign_tx(
         msg.data_initial_chunk = chunk
 
     response = client.call(msg)
+    assert isinstance(response, messages.EthereumTxRequest)
 
     while response.data_length is not None:
         data_length = response.data_length
+        assert data is not None
         data, chunk = data[data_length:], data[:data_length]
         response = client.call(messages.EthereumTxAck(data_chunk=chunk))
+        assert isinstance(response, messages.EthereumTxRequest)
+
+    assert response.signature_v is not None
+    assert response.signature_r is not None
+    assert response.signature_s is not None
 
     # https://github.com/trezor/trezor-core/pull/311
     # only signature bit returned. recalculate signature_v
     if response.signature_v <= 1:
+        assert chain_id is not None
```

Comment:
you shouldn't need this assert, given that we exclude `chain_id is None` at start?

Reply:
Done in b70acaf